dev/benchmarks/test_apps/stocks/lib/stock_types.dart

```diff
@@ -3,22 +3,23 @@
 // found in the LICENSE file.
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
 
 enum StockMode { optimistic, pessimistic }
 enum BackupMode { enabled, disabled }
 
-class StockConfiguration {
+class StockConfiguration extends ChangeNotifier {
```

Comment:
can you elaborate on your decision to make this mutable? it seems like it would be simpler overall to have an immutable class that gets passed around as data rather than having a pub/sub model.

Reply:
Because I want to use inheritedNotifier. Alternatively, I can make this immutable and use inheritedWidget to host it, but I will have to wrap the inheritedwidget with statefulwidget which requires some works too.

Reply:
wouldn't you inherently have something stateful around if there's immutable state that changes? Who owns the app state?